src/components/pages/AccountSettings/EditMember.tsx

```diff
@@ -0,0 +1,202 @@
+import * as React from "react";
+import { MapStateToProps, connect } from "react-redux";
+import { View, TextStyle, ViewStyle } from "react-native";
+import { withNavigation, NavigationInjectedProps } from "react-navigation";
+
+import { ApiEndpointName } from "@raha/api-shared/dist/routes/ApiEndpoint";
+import { MemberId } from "@raha/api-shared/dist/models/identifiers";
+
+import { RahaState } from "../../../store";
+import { Member } from "../../../store/reducers/members";
+import { getLoggedInMember } from "../../../store/selectors/authentication";
+import { KeyboardAwareScrollContainer } from "../../shared/elements/KeyboardAwareScrollContainer";
+import { Text, Button, TextInput } from "../../shared/elements";
+import { Loading } from "../../shared/Loading";
+import { styles as commonStyles } from "./styles";
+import { getUsername } from "../../../helpers/username";
+import { fonts } from "../../../helpers/fonts";
+import { colors } from "../../../helpers/colors";
+import { editMember } from "../../../store/actions/me";
+import {
+  ApiCallStatus,
+  ApiCallStatusType
+} from "../../../store/reducers/apiCalls";
+import { getStatusOfApiCall } from "../../../store/selectors/apiCalls";
+
+interface OwnProps {}
+
+interface StateProps {
+  loggedInMember: Member;
+  editMemberApiCallStatus?: ApiCallStatus;
+}
+
+interface DispatchProps {
+  editMember: (
+    loggedInMemberId: MemberId,
+    fullName?: string,
+    username?: string
+  ) => void;
+}
+
+type Props = OwnProps & StateProps & DispatchProps & NavigationInjectedProps;
+
+interface State {
+  username: string;
+  fullName: string;
+}
+
+class EditMemberPageView extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    const { loggedInMember } = this.props;
+    this.state = {
+      username: loggedInMember.get("username"),
+      fullName: loggedInMember.get("fullName")
+    };
+  }
+
+  canUpdate() {
+    const { loggedInMember } = this.props;
+    const currentFullName = loggedInMember.get("fullName");
+    const { fullName } = this.state;
+    if (!fullName || currentFullName === fullName) {
+      return false;
+    }
+    return true;
+  }
+
+  onChangeFullName = (newName: string) => {
+    const { loggedInMember } = this.props;
+    const currentUsername = loggedInMember.get("username");
+    const currentFullName = loggedInMember.get("fullName");
+
+    if (newName === currentFullName) {
+      this.setState({
```

Comment:
Does this need a special case? Is it for performance reasons?

Reply:
`getUsername` generates a new username, while this case sets the username back to the current username if the full name is the same as the current full name.

Reply:
How about something like
```
this.setState({
        fullName: newName,
        username: newName === currentFullName ? currentUsername : getUsername(newName)
      })
```
?

Reply:
lol what it's like you'd prefer we have nice code or something? derp ty lol XD